views/js/controller/runtime/testRunner.js

```diff
@@ -1307,7 +1312,17 @@ function (
 
                 $controls.$forwardButtons.click(function () {
                     if (!$(this).hasClass('disabled')) {
-                        TestRunner.moveForward();
+                        if (_isEmptyItemWarningModalEnabled()) {
+                            $doc
+                                .off('testRunner:moveForward')
+                                .on('testRunner:moveForward', function() {
```

Comment:
I am afraid the event name does not comply to our coding rules. Namespace separator is the dot `.`.  And usually we tend to use dashes as word separator, all in lowercase: `testrunner-moveforward`.